cloud-sql/mysql/Startup.cs

```diff
@@ -55,152 +54,157 @@ public void ConfigureServices(IServiceCollection services)
-            // "Uid=<DB_USER>;Pwd=<DB_PASS>;Host=<DB_HOST>;Database=<DB_NAME>;"
-            var connectionString = new MySqlConnectionStringBuilder()
+            public MySqlConnectionStringBuilder NewMysqlTCPConnectionString()
             {
-                // The Cloud SQL proxy provides encryption between the proxy and instance.
-                SslMode = MySqlSslMode.None,
+                // [START cloud_sql_mysql_dotnet_ado_connection_tcp]
+                // Equivalent connection string:
+                // "Uid=<DB_USER>;Pwd=<DB_PASS>;Host=<DB_HOST>;Database=<DB_NAME>;"
+                var connectionString = new MySqlConnectionStringBuilder()
+                {
+                    // The Cloud SQL proxy provides encryption between the proxy and instance.
+                    SslMode = MySqlSslMode.None,
 
-                // Remember - storing secrets in plaintext is potentially unsafe. Consider using
-                // something like https://cloud.google.com/secret-manager/docs/overview to help keep
-                // secrets secret.
-                Server = Environment.GetEnvironmentVariable("DB_HOST"),   // e.g. '127.0.0.1'
-                // Set Host to 'cloudsql' when deploying to App Engine Flexible environment
-                UserID = Environment.GetEnvironmentVariable("DB_USER"),   // e.g. 'my-db-user'
-                Password = Environment.GetEnvironmentVariable("DB_PASS"), // e.g. 'my-db-password'
-                Database = Environment.GetEnvironmentVariable("DB_NAME"), // e.g. 'my-database'
-            };
-            connectionString.Pooling = true;
-            // Specify additional properties here.
-            // [START_EXCLUDE]
-            SetDbConfigOptions(connectionString);
-            // [END_EXCLUDE]
-            DbConnection connection =
-                new MySqlConnection(connectionString.ConnectionString);
-            // [END cloud_sql_mysql_dotnet_ado_connection_tcp]
-            return connection;
-        }
+                    // Remember - storing secrets in plaintext is potentially unsafe. Consider using
+                    // something like https://cloud.google.com/secret-manager/docs/overview to help keep
+                    // secrets secret.
+                    Server = Environment.GetEnvironmentVariable("DB_HOST"),   // e.g. '127.0.0.1'
+                    // Set Host to 'cloudsql' when deploying to App Engine Flexible environment
+                    UserID = Environment.GetEnvironmentVariable("DB_USER"),   // e.g. 'my-db-user'
+                    Password = Environment.GetEnvironmentVariable("DB_PASS"), // e.g. 'my-db-password'
+                    Database = Environment.GetEnvironmentVariable("DB_NAME"), // e.g. 'my-database'
+                };
+                connectionString.Pooling = true;
+                // Specify additional properties here.
+                // [START_EXCLUDE]
+                SetDbConfigOptions(connectionString);
+                // [END_EXCLUDE]

```

Comment:
If you move this to the `GetMySqlConnectionString()` method, before returning either of the connection strings, then you can remove it from here and from lines 190-192. I've left the suggestion in `GetMySqlConnectionString()`.

Reply:
Done.